hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataReader.java

```diff
@@ -339,6 +341,32 @@ public List<OmKeyInfo> listKeys(String volumeName, String bucketName,
     }
   }
 
+  @Override
```

Comment:
nit: 
1. I think it would be nicer if you create a helper function that transformation `OmKeyInfo` to `BasicOmKeyInfo` and use the same in `OzoneManager` and `OmSnapshot`.  You can add this helper to `BasicOmKeyInfo` class itself.

```suggestion
  @Override
  public List<BasicOmKeyInfo> listKeysLight(String volumeName,
                                            String bucketName,
                                            String startKey,
                                            String keyPrefix,
                                            int maxKeys) throws IOException {

    List<OmKeyInfo> keys = listKeys(volumeName, bucketName, startKey, keyPrefix,
        maxKeys);

    return keys.stream().map(BasicOmKeyInfo::fromOmKeyInfo)
        .collect(Collectors.toList());
  }
```

And something like this in `BasicOmKeyInfo`:

```
  public static BasicOmKeyInfo fromOmKeyInfo(OmKeyInfo omKeyInfo) {
    return new BasicOmKeyInfo(
        omKeyInfo.getVolumeName(),
        omKeyInfo.getBucketName(),
        omKeyInfo.getKeyName(),
        omKeyInfo.getDataSize(),
        omKeyInfo.getCreationTime(),
        omKeyInfo.getModificationTime(),
        omKeyInfo.getReplicationConfig(),
        omKeyInfo.isFile());
  }
```

2. Also there is no need of comments you added. It is just repeating what code is coding. Code comment should be added only to state why something is done in particular way. And I don't think `why` is needed here because it is pretty straightforward.

Reply:
Understood, thank you @hemantk-12. Noted your comments and made the required changes.

Reply:
Thanks for incorporating the suggestions. I think code looks much cleaner now.